compiler/modules/types/list.bal

```diff
@@ -582,58 +582,44 @@ function listSubtypeDiff(Context cx, SubtypeData t1, SubtypeData t2) returns Sub
     return b;
 }
 
-function isDisjoint(Context cx, Bdd t1, Bdd t2) returns boolean {
-    ListAtomicType[]? posAtoms = bddListAtoms(cx, t1);
-    ListAtomicType[]? negAtoms = bddListAtoms(cx, t2);
-    if posAtoms is () || negAtoms is () {
-        return false;
+function isTupleDisjointSimple(Context cx, Bdd t1, Bdd t2) returns boolean? {
```

Comment:
Make it `listIsEmptySimple(Context cx, Bdd) returns boolean?`.

Reply:
Note that we can memoize the result when it's true.

Reply:
fixed